packages/dev/test/RelayServer.test.ts

```diff
@@ -605,6 +606,212 @@ contract('RelayServer', function (accounts: Truffle.Accounts) {
+      const serverSpy = sinon.spy(env.relayServer)
+      const txHashes = await env.relayServer.withdrawToOwnerIfNeeded(currentBlockNumber)
+      assert.deepEqual(txHashes, [])
+      sinon.assert.calledOnce(serverSpy.isReady)
+      assert.isTrue(serverSpy.isReady.returnValues[0])
+    })
+
+    describe('_resolveWithdrawalConfig', function () {
+      async function assertReturnedError (message: string): Promise<void> {
+        const spy = sinon.spy(env.relayServer.logger, 'error')
+        await env.relayServer.withdrawToOwnerIfNeeded(currentBlockNumber)
+        sinon.assert.calledWith(spy, `withdrawToOwnerIfNeeded: ${message}`)
+        spy.restore()
+      }
+      it('should throw with bad config file', async function () {
+        const badWithdrawConfig = JSON.stringify({
+          withdrawOnEthBalanceReached: 1e18,
+          leaveManagerWithAmountEth: 0.5e18,
+          repeat: false,
+          blah: 'blah'
+        })
+        fs.writeFileSync(configFilename, badWithdrawConfig)
+        try {
+          await env.relayServer._resolveWithdrawalConfig(configFilename)
+          assert.fail()
+        } catch (e) {
+          assert.include(e.message, 'Did not expect property `blah` to exist, got `blah` in object `withdrawConfig`')
+          await assertReturnedError(e.message)
+        }
+      })
+
+      it('should throw if withdrawOnEthBalanceReached < leaveManagerWithAmountEth', async function () {
+        const badWithdrawConfig = JSON.stringify({
+          withdrawOnEthBalanceReached: 0.5e18,
+          leaveManagerWithAmountEth: 1e18,
+          repeat: false
+        })
+        fs.writeFileSync(configFilename, badWithdrawConfig)
+        try {
+          await env.relayServer._resolveWithdrawalConfig(configFilename)
+          assert.fail()
+        } catch (e) {
+          assert.include(e.message, 'withdrawOnEthBalanceReached must be at least leaveManagerWithAmountEth')
+          await assertReturnedError(e.message)
+        }
+      })
+
+      it('should throw if leaveManagerWithAmountEth < managerTargetBalance', async function () {
+        const badWithdrawConfig = JSON.stringify({
+          withdrawOnEthBalanceReached: env.relayServer.config.managerTargetBalance * 10,
+          leaveManagerWithAmountEth: env.relayServer.config.managerTargetBalance / 2,
+          repeat: false
+        })
+        fs.writeFileSync(configFilename, badWithdrawConfig)
+        try {
+          await env.relayServer._resolveWithdrawalConfig(configFilename)
+          assert.fail()
+        } catch (e) {
+          assert.include(e.message, `leaveManagerWithAmountEth must be at least managerTargetBalance ${env.relayServer.config.managerTargetBalance}`)
+          await assertReturnedError(e.message)
+        }
+      })
+
+      it('should throw if withdrawOnEthBalanceReached - leaveManagerWithAmountEth < 1 ether  and config.smallAmount is not configured', async function () {
```

Comment:
too much configuration.
maybe have `minimumWithdrawEth` - which is the (minimum) value which will be sent to the owner. 
it will be sent only after the manager withdraw to itself, and leave its own balance high enough. so there is no need for the extra "small amount" flag: if I specify `minimumWithdrawEth = 0.1` then its MY problem...

Reply:
Ok